services/visual_recognition/v3.js

```diff
@@ -0,0 +1,362 @@
+        visualRecognition.deleteClassifier(params, cbdone);
+        break;
+    }
+  }
+
+  // This is the Watson Visual Recognition V3 Node
+  function WatsonVisualRecognitionV3Node (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+    var message;
+    this.on('input', function (msg) 
+    {
+      // Check which single feature has been requested.
+      var feature = config['image-feature'];
+      node.status({});
+      // so there is at most 1 temp file at a time (did not found a better solution...)
+      temp.cleanup(); 
+
+      if (!msg.payload) {
+        this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+        message = 'Missing property: msg.payload';
+        node.error(message, msg);
+        return;
+      }
+      if (feature !== 'retrieveClassifiersList' && feature !== 'retrieveClassifierDetails' 
+          && feature !== 'deleteClassifier')
+      {
+        if (typeof msg.payload === 'boolean' || typeof msg.payload === 'number') 
+        {
+          this.status({fill:'red', shape:'ring', text:'bad format payload'}); 
+          message = 'Bad format : msg.payload must be a URL string or a Node.js Buffer';
+          node.error(message, msg);
+          return;
+        }
+      }
+
+      // If it is present the newly provided user entered key 
+      // takes precedence over the existing one. 
+      apikey = s_apikey || this.credentials.apikey;
+      this.status({}); 
+
+      if (!apikey) {
+        this.status({fill:'red', shape:'ring', text:'missing credentials'});          
+        message ='Missing Watson Visual Recognition API service credentials'; 
+        node.error(message, msg);
+        return;
+      }
+
+      var params = {};
+
+      // This is the callback after the call to the watson service.    
+      // Set up as a var within this scope, so it has access to node, msg etc. 
+      // in preparation for the Watson service action       
+      var actionComplete = function(err, body, other) {
+
+        if (err != null && body==null)
+        {
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_code']= err.code;
+          if (!err.error)
+            msg.result['error']= err.error;
+          console.log('Error:', err.code);
+          node.error('Error code : ' +  err.code);
+          return;
+        }
+        else if (body.images[0].error)
+        {
+          var err_desc = body.images[0].error.description;
+          var err_id = body.images[0].error.error_id
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_id']= err_id;
+          msg.result['error']= err_desc;
+          console.log('Error:', err_desc);
+          msg.payload='see msg.result.error';
+          node.send(msg); 
+        }
+        else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l195)